analytics-integrations/moengage/src/main/java/com/segment/analytics/internal/integrations/MoEngageIntegration.java

```diff
@@ -0,0 +1,173 @@
+    }
+
+    @Override
+    public void onActivityResumed(Activity activity) {
+        super.onActivityResumed(activity);
+        mHelper.onResume(activity);
+    }
+
+    @Override
+    public void onActivityPaused(Activity activity) {
+        super.onActivityPaused(activity);
+        mHelper.onPause(activity);
+    }
+
+    @Override
+    public void onActivityStopped(Activity activity) {
+        super.onActivityStopped(activity);
+        mHelper.onStop(activity);
+    }
+
+    @Override
+    public void onActivitySaveInstanceState(Activity activity, Bundle outState) {
+        super.onActivitySaveInstanceState(activity, outState);
+        mHelper.onSaveInstanceState(outState);
+    }
+
+    @Override
+    public void identify(IdentifyPayload identify) {
+        super.identify(identify);
+        Traits tr = identify.traits();
+        if( null == tr || tr.isEmpty()){
+            return;
+        }
+        Set<String> keys = tr.keySet();
+        HashMap<String, Object> processedAttrs = new HashMap<String, Object>();
+        for(String key : keys){
+            if( null != key ){
+                if( ANONYMOUS_ID_KEY.equals(key)) {
+                    processedAttrs.put(USER_ATTR_SEGMENT_AID, tr.get(key));
+                } else if(EMAIL_KEY.equals(key)){
+                    processedAttrs.put(MoEHelperConstants.USER_ATTRIBUTE_USER_EMAIL, tr.get(key));
+                } else if(USER_ID_KEY.equals(key)){
+                    processedAttrs.put(MoEHelperConstants.USER_ATTRIBUTE_UNIQUE_ID, tr.get(key));
+                } else if(NAME_KEY.equals(key)){
+                    processedAttrs.put(MoEHelperConstants.USER_ATTRIBUTE_USER_NAME, tr.get(key));
+                } else if(PHONE_KEY.equals(key)){
+                    processedAttrs.put(MoEHelperConstants.USER_ATTRIBUTE_USER_MOBILE, tr.get(key));
+                } else if(FIRST_NAME_KEY.equals(key)){
+                    processedAttrs.put(MoEHelperConstants.USER_ATTRIBUTE_USER_FIRST_NAME, tr.get(key));
+                } else if(LAST_NAME_KEY.equals(key)){
+                    processedAttrs.put(MoEHelperConstants.USER_ATTRIBUTE_USER_LAST_NAME, tr.get(key));
+                } else if(GENDER_KEY.equals(key)) {
+                    processedAttrs.put(MoEHelperConstants.USER_ATTRIBUTE_USER_GENDER, tr.get(key));
+                } else if(BIRTHDAY_KEY.equals(key)) {
+                    DateFormat df = new SimpleDateFormat("MM/dd/yyyy");
+                    String birthDate = df.format(tr.birthday());
+                    processedAttrs.put(MoEHelperConstants.USER_ATTRIBUTE_USER_BDAY, birthDate);
+                } else {
+                    processedAttrs.put(key, tr.get(key));
+                }
+            }else if(MoEHelper.APP_DEBUG){
+                Log.e(MoEHelper.TAG, "TRAIT KEY CANNOT BE NULL");
+            }
+        }
+        if( hasLocationPermissions ){
+            AnalyticsContext.Location location = identify.context().location();
```

Comment:
This may be set manually by our users, so you don't need to check the location permission.